### ISSUE TITLE
Discussion: New package for flags parsing.

### ISSUE THREAD

**Issue by ChriMarMe** [bobuhiro11/gokvm#151]
I suggest (Kong)[https://github.com/alecthomas/kong] for a new commandline parser.

It will bring sophisticated commandline parsing and gives more future prove capabilities when extending gokvm.

**Comment by bobuhiro11** [comment]
Thanks for the suggestion.
But as a project policy I would like to develop only with the standard library.
I believe that is sufficient in many cases, but are there cases where that is not possible?

**Comment by ChriMarMe** [comment]
I can certainly think about extending gokvm to a more complex VMM with load/store/start/stop/resume functionality.
Decoupling the execution of the binary and running a VM which needs to be controlled via command execution. (see  [#150#issuecomment-1685713411](https://github.com/bobuhiro11/gokvm/pull/150#issuecomment-1685713411))

Have a look at the Draft PR #152 


On a more general note:
What do you think about working out a project plan for desired functionalities and capabilities?

**Comment by bobuhiro11** [comment]
I understand the intent to support operations like load/store/start/resume in addition to the existing VM booting. Interesting 👍 
I also understand the need for subcommand. Well, my suggestion is simply to prioritize the standard library. For example, I think we can achieve subcommands by doing the following:

https://gobyexample.com/command-line-subcommands

> On a more general note:
> What do you think about working out a project plan for desired functionalities and capabilities?

I would like to implement only essential aspects of virtualization in gokvm. For example, support for new boot protocols and load/store/start/resume functionality would be appropriate for gokvm. On the other hand, I don't want to include things that should be done by higher level services or orchestrators (OpenStack, Libvirt, etc.) such as listing or scheduling of multiple VMs.

**Comment by bobuhiro11** [comment]
Yes, I know that kong and a few others are really great cmdline parsers. And I also know that you want to spend your time on more essential work (e.g. PVH Boot), not cmdline parser. Sorry to put you through my selfishness. So, as a suggestion, how about using kong to implement subcommand for the time being, and then I will try the standard library? 

> Sorry to say: but in regards of commandline parsing I disagree with the desire to only use std libs.
> I dont want to invent the wheel again if we have a properly working one already.

I am focusing on using only the standard library since this project itself is a redevelopment of the wheel.

**Comment by ChriMarMe** [comment]
No need for excuses! I understand the desire to use std lib to keep robustness high.

This project may be a reinvention of the wheel but the only one which can be integrated into u-root and execute a VM from firmware :) So for me gokvm is not just pure interest.
